### ISSUE TITLE
member-ordering: method declaration mistreated as a member

### ISSUE THREAD

**Issue by msklvsk** [palantir/tslint#1156]
### Bug Report

- __TSLint version__: `3.8.0`
- __TypeScript version__: `1.8.10`
- __Running TSLint via__: Visual Studio Code

#### TypeScript code being linted

```ts
class Class {
  someMethod() { }

  // to be added in prototype at runtime
  mixinMethod: () => number;  // Error: Declaration of public instance member variable not allowed to appear after declaration of public instance member function
}
```

with `tslint.json`:

```json
{
  "rules": {
    "member-ordering": [true, "public-before-private", "static-before-instance", "variables-before-functions"]
  }
}
```

#### Actual behavior

tslint reports an error “Declaration of public instance member variable not allowed to appear after declaration of public instance member function” for `mixinMethod()`

#### Expected behavior

no error is reported

**Comment by adidahiya** [comment]
I believe this was fixed in https://github.com/palantir/tslint/pull/969. Can you confirm that VSCode is actually using TSLint 3.8 and not some older version it installed on its own?

**Comment by msklvsk** [comment]
Confirming this for v3.8.1, command line.
The variable isn't actually initialized inside the class, only type-declared. That's a mixin pattern from the official TS docs: http://www.typescriptlang.org/docs/handbook/mixins.html. Their samples also reproduce this.

**Comment by aN0obIs** [comment]
I have small issue with this new behavior.
Please, take a look at following code.

```
class Foo {
  public readonly bar = pureComputed(this._getBar); // this._getBar undefined here
  // other stuff
  private _getBar = () => { 
       // calc bar and return it
   }
}
```

Member ordering rule foces me to move property "bar" before arrow function that calculates its actual value. And when code compiles and runs I got error.
I know workarounds, but all them isn't very nice.
What would you suggest to me?

**Comment by mgol** [comment]
@aN0obIs I created a separate issue for this problem: #4156.
